### PR TITLE
add ISIS_AUTH_KEY parameter in ndfc_create_fabric.j2

### PR DESCRIPTION
**Pull request by ccoueffe**
## Related Collection Role

* [ ] cisco.nac_dc_vxlan.validate
* [x] cisco.nac_dc_vxlan.dtc.create
* [ ] cisco.nac_dc_vxlan.dtc.deploy
* [ ] cisco.nac_dc_vxlan.dtc.remove
* [ ] other

## Related Data Model Element

* [ ] vxlan.global
* [ ] vxlan.topology
* [x] vxlan.underlay
* [ ] vxlan.overlay_services
* [ ] vxlan.overlay_extensions
* [ ] vxlan.policy
* [ ] defaults.vxlan
* [ ] other

## Proposed Changes

Add Key ISIS_AUTH_KEY in template ndfc_create_fabric.j2

## Cisco NDFC Version

NDFC: 12.2.1.316

## Checklist

* [ ] Latest commit is rebased from develop with merge conflicts resolved
* [ ] New or updates to documentation has been made accordingly
* [ ] Assigned the proper reviewers

### CODE REVIEW

**Comment by ccoueffe** [comment]
@mtarking I added the missing parameter for ISIS authentication (underlay).

File: roles/dtc/common/templates/ndfc_create_fabric.j2 [hunk 1]
```diff
@@ -145,6 +145,7 @@
 {% endif %}
   ISIS_P2P_ENABLE: {{ underlay.isis.network_point_to_point | default(defaults.vxlan.underlay.isis.network_point_to_point) }}
 {% if (underlay.bfd.enable | default(defaults.vxlan.underlay.bfd.enable) | title) == 'True' %}
+  ISIS_AUTH_KEY: {{ underlay.isis.authentication_key }}
```
Comment (mikewiebe) [inline roles/dtc/common/templates/ndfc_create_fabric.j2:148]: Do we need a default value?

**Comment by ccoueffe** [comment]
Adding rule:

```yaml
---
vxlan:
  underlay:
    isis:
      level: level-2
      network_point_to_point: true
      authentication_enable: true
      authentication_key_id: 127
      authentication_keychain_name: mykey
      authentication_key: 070c285f4d0648574446
      overload_bit: true
      overload_bit_elapsed_time: 60
```

```bash
❯ iac-validate -s schemas/schema.yaml -r collections/ansible_collections/cisco/nac_dc_vxlan/roles/validate/files/rules/required_rules  host_vars/nac-ndfc2
ERROR - Semantic error, rule 203: Verify Fabric Underlay ISIS Authentication (['For vxlan.underlay.isis.authentication_enable is enabled, vxlan.underlay.isis.authentication_keychain_name must be provided'])

❯ iac-validate -s schemas/schema.yaml -r collections/ansible_collections/cisco/nac_dc_vxlan/roles/validate/files/rules/required_rules  host_vars/nac-ndfc2
ERROR - Semantic error, rule 203: Verify Fabric Underlay ISIS Authentication (['For vxlan.underlay.isis.authentication_enable is enabled, vxlan.underlay.isis.authentication_key must be provided'])
```

File: roles/dtc/common/templates/ndfc_create_fabric.j2 (hunk 1 above)
Comment (ccoueffe) [reply roles/dtc/common/templates/ndfc_create_fabric.j2]: No default expected. Customer should provided their own key. If key is not provided and auth is enabled, syntax validation  will fail.

I moved  Key in the if condition.

**Review by juburnet — APPROVED** [review]
Approved.  Functions correctly.
```
TASK [cisco.nac_dc_vxlan.validate : Perform Required Syntax and Semantic Model Validation] ******************************************************************************************************************************
Tuesday 29 October 2024  09:42:53 -0400 (0:00:00.050)       0:00:01.694 ******* 
Tuesday 29 October 2024  09:42:53 -0400 (0:00:00.050)       0:00:01.694 ******* 
Semantic error, rule 203: Verify Fabric Underlay ISIS Authentication (['For vxlan.underlay.isis.authentication_enable is enabled, vxlan.underlay.isis.authentication_key must be provided', 'For vxlan.underlay.isis.authentication_enable is enabled, vxlan.underlay.isis.authentication_keychain_name must be provided'])
```

**Comment by juburnet** [comment]
@mikewiebe this is ready

File: roles/dtc/common/templates/ndfc_create_fabric.j2 [hunk 2]
```diff
@@ -138,6 +138,7 @@
 {% if (underlay.isis.authentication_enable | default(defaults.vxlan.underlay.isis.authentication_enable) | title) == 'True' %}
   ISIS_AUTH_KEYCHAIN_NAME: {{ underlay.isis.authentication_keychain_name | default(omit) }}
   ISIS_AUTH_KEYCHAIN_KEY_ID: {{ underlay.isis.authentication_key_id | default(defaults.vxlan.underlay.isis.authentication_key_id) }}
+  ISIS_AUTH_KEY: {{ underlay.isis.authentication_key }}
```
Comment (mikewiebe) [inline roles/dtc/common/templates/ndfc_create_fabric.j2:141]: Ok just to make sure I understand.  If `authentication_key` is not set in the data model then the rule below will catch it right?

**Review by mikewiebe — APPROVED** [review]

**Pull request merged**